checker/src/main/java/org/checkerframework/checker/initialization/InitializationParentAnnotatedTypeFactory.java

```diff
@@ -716,7 +716,7 @@ public Void visitExecutable(AnnotatedExecutableType t, Void p) {
             if (elem.getKind() == ElementKind.CONSTRUCTOR) {
                 AnnotatedDeclaredType returnType = (AnnotatedDeclaredType) t.getReturnType();
                 DeclaredType underlyingType = returnType.getUnderlyingType();
-                returnType.replaceAnnotation(
+                returnType.addMissingAnnotation(
```

Comment:
I changed this to only add a missing annotation.
Can you expand the tests in `checker/jtreg/nullness/constructor-initialization` to add a non-checked file that uses an explicit initialization annotation and then invoke that constructor from a checked class? We should make sure it behaves as intended.